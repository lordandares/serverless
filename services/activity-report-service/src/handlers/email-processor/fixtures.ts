export const TZ_LA = 'America/Los_Angeles'
export const TZ_NYC = 'America/New_York'

export const MOCK_APPLICATIONS = [
  { _id: '111111111111111111111111', name: 'Application 1' },
  { _id: '222222222222222222222222', name: 'Application 2' },
]

export const MOCK_DATA = [
  {
    _id: '565fe59257c5d59e60cc35a1',
    application: '111111111111111111111111',
    email: 'testing@lighthouse.io',
    firstName: 'Testing',
    preferences: {
      notifications: {
        channels: [
          {
            enabled: true,
            name: 'daily-location-report',
            options: {
              hours: 9,
              timezone: TZ_LA,
            },
          },
        ],
      },
    },
    permissions: [
      {
        module: 'location',
        type: 'document',
        value: '565fe59257c5d59e60cc35b3',
      },
    ],
  },
  {
    _id: '565fe59257c5d59e60cc35a2',
    application: '111111111111111111111111',
    email: 'testing2@lighthouse.io',
    firstName: 'Testing2',
    preferences: {
      notifications: {
        channels: [
          {
            enabled: true,
            name: 'daily-location-report',
            options: {
              hours: 9,
              timezone: TZ_LA,
            },
          },
        ],
      },
    },
  },
  {
    _id: '565fe59257c5d59e60cc35a3',
    application: '222222222222222222222222',
    email: 'testing2@lighthouse.io',
    firstName: 'Testing 2',
    preferences: {
      notifications: {
        channels: [
          {
            enabled: true,
            name: 'daily-location-report',
            options: {
              hours: 11,
              timezone: TZ_LA,
            },
          },
        ],
      },
    },
    permissions: [
      {
        module: 'location',
        type: 'document',
        value: '565fe59257c5d59e60cc35b5',
      },
    ],
  },
]

export const MOCK_LOCATIONS = [
  {
    _id: '565fe59257c5d59e60cc35b3',
    files: {
      pdf: {
        path:
          'activity-reports/565fe59257c5d59e60cc35b3/1999/12/1999-12-31-1548194730622.pdf',
        timestamp: new Date('2019-12-31T18:00:00Z'),
      },
    },
    name: 'Testing Location 1',
  },
  {
    _id: '565fe59257c5d59e60cc35b4',
    files: {
      pdf: {
        path:
          'activity-reports/565fe59257c5d59e60cc35b4/1999/12/1999-12-30-1548194730622.pdf',
        timestamp: new Date('2019-12-30T18:00:00Z'),
      },
    },
    name: 'Testing Location 2',
  },
  {
    _id: '565fe59257c5d59e60cc35b5',
    files: {
      pdf: {
        path:
          'activity-reports/565fe59257c5d59e60cc35b5/1999/12/1999-12-29-1548194730622.pdf',
        timestamp: new Date('2019-12-29T18:00:00Z'),
      },
    },
    name: 'Testing Location 3',
    properties: {
      area: '565fe59257c5d59e60cc35c1',
    },
  },
]

export const MOCK_USER_DETAILS = {
  email: 'testing@lighthouse.io',
  firstName: 'Testing',
}

export const MOCK_EMAIL = {
  Destination: {
    CcAddresses: [],
    ToAddresses: ['testing@test.com'],
  },
  ReplyToAddresses: ['support@lighthouse.io'],
  Source: 'support@lighthouse.io',
  Template: 'template-name',
  TemplateData: '',
}

export const MOCK_RESULT1 = {
  ResponseMetadata: {
    RequestId: '88464904-fcf3-11e8-8e1f-b9c0f7c86ac5',
  },
  MessageId: '010001679b4692cd-28db95eb-8524-4abf-9303-9a1f32824a2f-000000',
}

export const MOCK_RESULT2 = {
  ResponseMetadata: {
    RequestId: '88464904-fcf3-11e8-8e1f-b9c0f7c86ac6',
  },
  MessageId: '010001679b4692cd-28db95eb-8524-4abf-9303-9a1f32824a2f-000001',
}
