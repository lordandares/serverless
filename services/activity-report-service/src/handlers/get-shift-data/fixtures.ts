export const body = {
  shiftId: '1234567890',
}

export const data = {
  applicationId: '9c89816ea008a66615a220d1',
  audits: [{ _id: '5b36816ea008a70015a330d1' }],
  events: [{ _id: '5b36816ea008a70015a330d2' }],
  issues: [{ _id: '5b36816ea008a70015a330d3' }],
  location: {},
  shift: {},
  tasks: [{ _id: '5b36816ea008a70015a330d4' }],
  timestamp: '2018-11-29T00:25:19.020Z',
  timezone: 'Australia/Melbourne',
  user: {},
  zones: [{ _id: '5b36816ea008a70015a330d5' }],
}

export const emptyData = {
  ...data,
  audits: [],
  events: [],
  issues: [],
  tasks: [],
}

export const summaryS3Path =
  '123456789/pdfs/2018-11-26/1543450822684-dar-123456789.pdf'

export const formS3Path = [
  '123456789/pdfs/2018-11-26/1543450822683-tasks-12345.pdf',
  '123456789/pdfs/2018-11-26/1543450822683-audit-12345.pdf',
  '123456789/pdfs/2018-11-26/1543450822683-issue-12345.pdf',
]

export const references = {
  formS3Paths: [
    '123456789/pdfs/2018-11-26/1543450822683-tasks-12345.pdf',
    '123456789/pdfs/2018-11-26/1543450822683-audit-12345.pdf',
    '123456789/pdfs/2018-11-26/1543450822683-issue-12345.pdf',
  ],
  summaryS3Path: '123456789/pdfs/2018-11-26/1543450822684-dar-123456789.pdf',
}

export const referencesShouldNotSkip = {
  mergedS3Path:
    'activity-reports/shifts/9c89816ea008a66615a220d1/2000/01/01/2000-01-01-shift-1234567890-summary-and-forms-946684800000.pdf',
  pdfsToMerge: [
    '123456789/pdfs/2018-11-26/1543450822684-dar-123456789.pdf',
    '123456789/pdfs/2018-11-26/1543450822683-tasks-12345.pdf',
    '123456789/pdfs/2018-11-26/1543450822683-audit-12345.pdf',
    '123456789/pdfs/2018-11-26/1543450822683-issue-12345.pdf',
  ],
  skip: false,
  summaryS3Path: '123456789/pdfs/2018-11-26/1543450822684-dar-123456789.pdf',
}
