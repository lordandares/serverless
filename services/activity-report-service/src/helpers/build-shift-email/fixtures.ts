export const STAGE = 'production'

export const active_link = {
  id: '565fe59257c5d59e60cc35a2',
  link: 'http://link.com',
}

export const inactive_link = {
  id: '565fe59257c5d59e60cc35a3',
  link: null,
}

export const long_active_links_1 = {
  id: '565fe59257c5d59e60cc35a6',
  link: 'http://link.com',
  fullName: 'Steeve Jobs',
  timestamp: null,
  lastName: 'Jobs',
}

export const long_active_links_2 = {
  id: '565fe59257c5d59e60cc35a6',
  link: 'http://link.com',
  fullName: 'John Doe',
  timestamp: null,
  lastName: 'Doe',
}

export const long_active_links_3 = {
  id: '565fe59257c5d59e60cc35a6',
  link: 'http://link.com',
  fullName: 'Paul Allen',
  timestamp: null,
  lastName: 'Allen',
}

export const formSubmission = {
  id: '565fe59257c5d59e60cc35a3',
  link:
    'activity-reports/shifts/565e42d3d4c628373ab25231/2019/10/10/2019-10-10-shift-5d9f83bd8bb82c0001cc20c3-summary-1570749390473.pdf',
}

export const withoutFormSubmission = {
  id: '565fe59257c5d59e60cc35a3',
  link:
    'activity-reports/shifts/565e42d3d4c628373ab25231/2019/10/10/2019-10-10-shift-5d9f83bd8bb82c0001cc20c3-summary-and-forms-1570749390473.pdf',
}

export const user_details_summary = {
  email: 'testing@test.com',
  firstName: 'Steve',
  shifts: [
    {
      _id: '565fe59257c5d59e60cc35a2',
    },
    {
      _id: '565fe59257c5d59e60cc35a3',
    },
  ],
  preferences: {
    notifications: {
      channels: [
        {
          enabled: true,
          name: 'daily-shift-report',
          options: {
            hours: 9,
            timezone: 'America/Los_Angeles',
            formSubmissions: false,
          },
        },
      ],
    },
  },
  timezone: 'America/Los_Angeles',
}

export const long_user_details_summary = {
  email: 'testing@test.com',
  firstName: 'Steve',
  shifts: [
    {
      _id: '565fe59257c5d59e60cc35a2',
    },
    {
      _id: '565fe59257c5d59e60cc35a6',
    },
    {
      _id: '565fe59257c5d59e60cc35a9',
    },
  ],
  preferences: {
    notifications: {
      channels: [
        {
          enabled: true,
          name: 'daily-shift-report',
          options: {
            hours: 9,
            timezone: 'America/Los_Angeles',
            formSubmissions: false,
          },
        },
      ],
    },
  },
  timezone: 'America/Los_Angeles',
}

export const user_details_summary_and_forms = {
  ...user_details_summary,
  preferences: {
    notifications: {
      channels: [
        {
          enabled: true,
          name: 'daily-shift-report',
          options: {
            hours: 9,
            timezone: 'America/Los_Angeles',
            formSubmissions: true,
          },
        },
      ],
    },
  },
}
