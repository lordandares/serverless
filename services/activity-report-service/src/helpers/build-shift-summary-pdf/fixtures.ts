export const AWS_REGION = 'ap-southeast-2'
export const DEFINITION = { content: ['definition'] }
export const LOGO_URL = 'http://lighthouse.io/logo.png'

export const CLOUDINARY_BASE_URL = 'http://cloudinary.com'
export const S3_BASE_URL = 'http://upload-test.com'
export const S3_BUCKET_UPLOADS = 'testing'
export const SUMMARY_REFERENCE =
  'activity-reports/shifts/111122223333/2000/01/01/2000-01-01-shift-0987654321-summary-946684800000.pdf'

export const DATA = {
  applicationId: '111122223333',
  audits: [
    {
      createdAt: '2000-01-01T15:00:00.000Z',
      files: {
        pdf: {
          path: '123456789/pdfs/2018-11-26/1543450822683-audit-12345.pdf',
        },
      },
    },
  ],
  location: {
    _id: '123456789',
  },
  issues: [
    {
      createdAt: '2000-01-01T16:00:00.000Z',
      files: {
        pdf: {
          path: '123456789/pdfs/2018-11-26/1543450822683-issue-12345.pdf',
        },
      },
    },
  ],
  tasks: [
    {
      createdAt: '2000-01-01T09:00:00.000Z',
      files: {
        pdf: {
          path: '123456789/pdfs/2018-11-26/1543450822683-tasks-12345.pdf',
        },
      },
    },
  ],
  timestamp: '2000-01-01T12:00:00.000Z',
  timezone: 'Australia/Melbourne',
  shift: {
    _id: '0987654321',
  },
}

export const BUILD_DATA = {
  data: {
    ...DATA,
    maxScans: 100,
    settings: {
      awsS3BaseUrl: S3_BASE_URL,
      cloudinaryBaseUrl: CLOUDINARY_BASE_URL,
    },
  },
  pdfOptions: {
    logoUrl: LOGO_URL,
    pageSize: 'A4',
  },
}

export const { data: CONFIG_DATA, pdfOptions: PDF_OPTIONS } = BUILD_DATA
