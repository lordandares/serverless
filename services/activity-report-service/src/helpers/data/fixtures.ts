export const payload = {
  shiftId: '5c18e2d030148e00017e9bdf',
}

export const secretId = 'lio/serverless/dar-service/secretId'

export const application = '5bff0da28d7c870001a01639'

export const audits = [
  {
    _id: '5bf73690dba3c200011dd416',
    createdAt: '2019-09-19T18:06:56.477+0000',
    title: 'Health and Safety',
    zone: '5bb7d1c618c1bb0001e8000a',
  },
  {
    _id: '5bf73690dba3c200011dd417',
    createdAt: '2019-09-19T19:07:56.477+0000',
    title: 'Medical First Aid Kits',
    area: {
      label: 'Emergency Room',
    },
  },
  {
    _id: '5bf73690dba3c200011dd418',
    createdAt: '2019-09-19T20:08:56.477+0000',
    title: 'Trip Hazards',
    gps: {
      reverseGeocoded: {
        label: 'Broadway, NYC, New York',
      },
    },
  },
]

export const events = [
  {
    _id: '5bf739c4dba3c200011de26c',
    timestamp: '2019-09-19T11:45:00.000+0000',
    zone: '5bb7d1c618c1bb0001e8000b',
    type: 'enter',
  },
  {
    _id: '5bf739c4dba3c200011de26d',
    timestamp: '2019-09-19T11:50:00.000+0000',
    zone: '5bb7d1c618c1bb0001e8000b',
    type: 'enter',
  },
  {
    _id: '5bf739c4dba3c200011de26e',
    timestamp: '2019-09-19T12:15:00.000+0000',
    zone: '5bb7d1c618c1bb0001e8000b',
    type: 'enter',
  },
  {
    _id: '5bf739c4dba3c200011de26g',
    geometryReverseGeocoded: {
      label: 'Broadway, NYC, New York',
    },
    timestamp: '2019-09-19T12:15:30.000+0000',
    type: 'geo',
    zone: undefined,
  },
]

export const issues = [
  {
    _id: '5bf73690dba3c200011dd416',
    createdAt: '2019-09-19T22:06:56.477+0000',
    entry: {
      summaryFields: [
        {
          fieldtype: 'text',
          label: 'Activity Type',
          value: 'Parking Area Check',
        },
        {
          fieldtype: 'text',
          label: 'Activity Details',
          value:
            'Officer Blake: Darcars College Park Nissan is officially safe and secured, no problems in disturbances or incidents, site is complete.',
        },
        {
          fieldtype: 'list',
          label: 'Photo',
          options: { type: 'media' },
          value: [
            '59e836b04260960f86301471/324873FE-ABE5-4795-9187-ED2DB3F34BA8-6B7DDF4A-8E8A-4DD9-AE54-E8CCAA53B23A.jpg',
            '59e836b04260960f86301471/324873FE-ABE5-4795-9187-ED2DB3F34BA8-95474B07-5826-4AA5-B4B7-1CEF2F533F20.jpg',
            '59e836b04260960f86301471/324873FE-ABE5-4795-9187-ED2DB3F34BA8-E4E8AB2D-A3F1-462A-98ED-C3106C975A31.jpg',
            '59e836b04260960f86301471/324873FE-ABE5-4795-9187-ED2DB3F34BA8-9E78D6B0-BDC9-427D-942A-E6D5CB0466C5.jpg',
          ],
        },
        {
          fieldtype: 'text',
          label: 'Activity Type',
          value: 'Parking Area Check Backyard',
        },
        {
          fieldtype: 'text',
          label: 'Activity Details',
          value:
            'Officer Blake: Darcars College Backyard is officially safe and secured.',
        },
        {
          fieldtype: 'list',
          label: 'Photo',
          options: { type: 'media' },
          value: [],
        },
      ],
    },
    title: 'Incident Report (Internal Patrols)',
    zone: '5bb7d1c618c1bb0001e8000b',
  },
  {
    _id: '5bf73690dba3c200011dd419',
    createdAt: '2019-09-19T23:09:56.477+0000',
    entry: {
      summaryFields: [
        {
          fieldtype: 'text',
          label: 'Activity Type',
          value: 'Building Area Check',
        },
        {
          fieldtype: 'text',
          label: 'Activity Details',
          value: 'Building patrol monitoring at Empire State.',
        },
      ],
    },
    gps: {
      reverseGeocoded: {
        label: 'Empire State, NYC, New York',
      },
    },
    title: 'Building Patrol',
  },
]

export const location = {
  _id: '5bf73690dba3c200011dd888',
  application: '5bff0da28d7c870001a01639',
  name: 'Testing Location',
  timezone: 'America/La_Paz',
}

export const locations = [
  {
    _id: '5bf73690dba3c200011dd888',
    name: 'Lighthouse Collingwood',
  },
]

export const locationWithArea = {
  ...location,
  properties: {
    area: '5bff0da28d7c870001a01639',
  },
}

export const area = {
  _id: locationWithArea.properties.area,
  geometry: {
    coordinates: [
      [
        [144.98665809631345, -37.79712984141311],
        [144.98663127422333, -37.79726760509564],
        [144.98681366443634, -37.79729303836279],
        [144.9868431687355, -37.797148916399856],
        [144.98665809631345, -37.79712984141311],
      ],
    ],
    type: 'Polygon',
  },
}

export const tasks = [
  {
    _id: '5bf73690dba3c200011dd416',
    createdAt: '2019-09-19T10:06:56.477+0000',
    entry: {
      summaryFields: [
        {
          fieldtype: 'text',
          label: 'Activity Type',
          value: 'Shopping Area Check',
        },
        {
          fieldtype: 'text',
          label: 'Activity Details',
          value:
            'Shopping patrol monitoring at Walmart. No suspicious behavior to report.',
        },
        {
          fieldtype: 'list',
          label: 'Photo',
          options: { type: 'media' },
          value: [
            '59e836b04260960f86301471/324873FE-ABE5-4795-9187-ED2DB3F34BA8-6B7DDF4A-8E8A-4DD9-AE54-E8CCAA53B23A.jpg',
          ],
        },
        {
          fieldtype: 'text',
          label: 'Activity Type',
          value: 'Parking Area Check',
        },
        {
          fieldtype: 'text',
          label: 'Activity Details',
          value:
            'Shopping patrol monitoring at Walmart Parking. No suspicious behavior to report.',
        },
        {
          fieldtype: 'list',
          label: 'Photo',
          options: { type: 'media' },
          value: [],
        },
      ],
    },
    title: 'Clean Office',
    zone: '5bb7d1c618c1bb0001e8000b',
  },
]

export const timestamp = '2019-09-19T20:30:00.000+0000'
export const timezone = 'America/La_Paz'

export const settings = {
  awsS3BaseUrl: 'https://uploads-lighthouse-io.s3.amazonaws.com',
  cloudinaryBaseUrl: 'https://res.cloudinary.com/lighthouseio/image/fetch',
}

export const user = {
  _id: '5bb7d1c618c1bb0001e8028a',
  firstName: 'John',
  lastName: 'Doe',
}

export const zones = [
  {
    _id: '5bb7d1c618c1bb0001e8000a',
    name: 'Blue Hill Command Center',
  },
  {
    _id: '5bb7d1c618c1bb0001e8000b',
    name: 'Under Armour',
  },
]

export const shift = {
  _id: '5c18e2d030148e00017e9bdf',
  breaks: [
    {
      end: {
        time: '2019-09-19T15:00:00.000+0000',
        zone: '5bb7d1c618c1bb0001e8000a',
      },
      start: {
        time: '2019-09-19T14:40:00.000+0000',
        zone: '5bb7d1c618c1bb0001e8000a',
      },
    },
    {
      end: {
        time: '2019-09-19T17:30:00.000+0000',
        zone: '5bb7d1c618c1bb0001e8000a',
      },
      start: {
        time: '2019-09-19T17:00:00.000+0000',
        zone: '5bb7d1c618c1bb0001e8000a',
      },
    },
  ],
  duration: 30000000,
  end: {
    time: '2019-09-19T20:00:00.000+0000',
    zone: '5bb7d1c618c1bb0001e8000a',
  },
  start: {
    time: '2019-09-19T14:00:00.000+0000',
    zone: '5bb7d1c618c1bb0001e8000b',
  },
  user: '5bb7d1c618c1bb0001e8028a',
}
