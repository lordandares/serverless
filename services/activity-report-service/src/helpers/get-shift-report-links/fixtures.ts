import * as dateFns from 'date-fns'
import * as MockDate from 'mockdate'

MockDate.set('2000-01-01T00:00:00.000Z')
const timestamp = dateFns.subHours(new Date('2019-11-07T11:15:00Z'), 6)

export const SUMMARY_REPORT_PATH =
  'activity-reports-shifts/7af476e9c68b751112e38cb8/2019/10/10/2019-10-10-shift-5be365d9c68b740001d27ba7-summary-1548194730622.pdf'

export const SUMMARY_REPORT_AND_FORMS_PATH =
  'activity-reports-shifts/7af476e9c68b751112e38cb8/2019/10/10/2019-10-10-shift-5be365d9c68b740001d27ba7-summary-and-forms-1447283621511.pdf'

export const S3_BASE_URL = 'http://baseurl.com'

export const shift = {
  _id: '5be365d9c68b740001d27ba7',
  user: '5be365d9c68b740001d20001',
  files: {
    activityReportSummary: {
      path: SUMMARY_REPORT_PATH,
      timestamp,
    },
    activityReportSummaryAndForms: {
      path: SUMMARY_REPORT_AND_FORMS_PATH,
      timestamp,
    },
  },
}

export const user = {
  _id: '5be365d9c68b740001d20001',
  firstName: 'Steve',
  lastName: 'Jobs',
  files: {
    activityReportSummary: {
      path: SUMMARY_REPORT_PATH,
      timestamp,
    },
    activityReportSummaryAndForms: {
      path: SUMMARY_REPORT_AND_FORMS_PATH,
      timestamp,
    },
  },
}
