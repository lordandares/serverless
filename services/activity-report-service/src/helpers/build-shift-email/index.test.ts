const mockGetShiftReportLinks = jest.fn()
const datetime = new Date('2019-11-07T11:15:00Z')

jest.mock('../get-shift-report-links', () => ({
  getShiftReportLinks: mockGetShiftReportLinks,
}))

import * as MockDate from 'mockdate'

import { buildShiftEmail } from './'
import * as mockData from './fixtures'

describe('helpers:buildShiftEmail', () => {
  beforeEach(() => {
    process.env.SES_CONFIGURATION_SET_NAME = ''
    process.env.STAGE = mockData.STAGE
    MockDate.set('2000-01-01T13:00:00.000Z')
  })

  afterEach(() => {
    jest.clearAllMocks()
    MockDate.reset()
  })

  it('handles error and returns false', async () => {
    const error = new Error('Shift error')
    mockGetShiftReportLinks.mockRejectedValue(error)

    const result = await buildShiftEmail({
      userDetails: mockData.user_details_summary_and_forms,
      datetime,
    })

    expect(result).toEqual(false)
  })

  it('it returns email data', async () => {
    mockGetShiftReportLinks
      .mockResolvedValueOnce(mockData.active_link)
      .mockResolvedValue(mockData.inactive_link)

    const result = await buildShiftEmail({
      userDetails: mockData.user_details_summary_and_forms,
      datetime,
    })

    expect(mockGetShiftReportLinks).toHaveBeenCalledTimes(2)
    expect(mockGetShiftReportLinks).toHaveBeenCalledWith({
      datetime,
      shift: {
        _id: '565fe59257c5d59e60cc35a3',
      },
      withFormSubmissions: true,
    })
    expect(mockGetShiftReportLinks).toHaveBeenCalledWith({
      datetime,
      shift: {
        _id: '565fe59257c5d59e60cc35a2',
      },
      withFormSubmissions: true,
    })

    expect(result).toMatchSnapshot()
    expect.assertions(4)
  })

  it('it returns email data ordered by lastName and Timestamp', async () => {
    mockGetShiftReportLinks
      .mockResolvedValueOnce(mockData.long_active_links_1)
      .mockResolvedValueOnce(mockData.long_active_links_2)
      .mockResolvedValue(mockData.long_active_links_3)

    const result = await buildShiftEmail({
      userDetails: mockData.long_user_details_summary,
      datetime,
    })

    expect(mockGetShiftReportLinks).toHaveBeenCalledTimes(3)
    expect(mockGetShiftReportLinks).toHaveBeenCalledWith({
      datetime,
      shift: {
        _id: '565fe59257c5d59e60cc35a9',
      },
      withFormSubmissions: false,
    })
    expect(mockGetShiftReportLinks).toHaveBeenCalledWith({
      datetime,
      shift: {
        _id: '565fe59257c5d59e60cc35a6',
      },
      withFormSubmissions: false,
    })

    expect(mockGetShiftReportLinks).toHaveBeenCalledWith({
      datetime,
      shift: {
        _id: '565fe59257c5d59e60cc35a9',
      },
      withFormSubmissions: false,
    })

    expect(result).toMatchSnapshot()
    expect.assertions(5)
  })

  it('it returns email data with formSubmissions', async () => {
    mockGetShiftReportLinks.mockResolvedValueOnce(mockData.formSubmission)

    const result = await buildShiftEmail({
      userDetails: mockData.user_details_summary_and_forms,
      datetime,
    })

    expect(mockGetShiftReportLinks).toHaveBeenCalledWith({
      datetime,
      shift: {
        _id: '565fe59257c5d59e60cc35a2',
      },
      withFormSubmissions: true,
    })

    expect(result).toMatchSnapshot()
    expect.assertions(2)
  })

  it('it returns email data without formSubmissions', async () => {
    mockGetShiftReportLinks.mockResolvedValueOnce(
      mockData.withoutFormSubmission,
    )

    const result = await buildShiftEmail({
      userDetails: mockData.user_details_summary,
      datetime,
    })

    expect(mockGetShiftReportLinks).toHaveBeenCalledWith({
      datetime,
      shift: {
        _id: '565fe59257c5d59e60cc35a3',
      },
      withFormSubmissions: false,
    })

    expect(result).toMatchSnapshot()
    expect.assertions(2)
  })

  it('it returns email data with configuration set when the environment var is set', async () => {
    process.env.SES_CONFIGURATION_SET_NAME = 'ses-configuration-set'
    mockGetShiftReportLinks
      .mockResolvedValueOnce(mockData.active_link)
      .mockResolvedValue(mockData.inactive_link)

    const result = await buildShiftEmail({
      userDetails: mockData.user_details_summary_and_forms,
      datetime,
    })

    expect(result).toMatchSnapshot()
  })

  describe('when not running in production', () => {
    beforeEach(() => {
      process.env.STAGE = 'testing'
    })

    it('it sets isProduction as false, adds stage to subject line and uses test template', async () => {
      mockGetShiftReportLinks.mockResolvedValueOnce(mockData.active_link)

      const result = await buildShiftEmail({
        userDetails: mockData.user_details_summary_and_forms,
        datetime,
      })
      expect(result).toMatchSnapshot()
    })
  })

  describe('when no active or inactive shift', () => {
    it('it returns email data with hasNoShifts set to true', async () => {
      const userDetails = {
        ...mockData.user_details_summary_and_forms,
        shifts: [],
      }

      const result = await buildShiftEmail({
        userDetails,
        datetime,
      })
      expect(result).toMatchSnapshot()
    })
  })
})
