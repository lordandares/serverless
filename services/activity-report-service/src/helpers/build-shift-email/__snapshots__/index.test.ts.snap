// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helpers:buildShiftEmail it returns email data 1`] = `
Object {
  "Destination": Object {
    "CcAddresses": Array [],
    "ToAddresses": Array [
      "testing@test.com",
    ],
  },
  "ReplyToAddresses": Array [
    "notifications@lighthouse.io",
  ],
  "Source": "notifications@lighthouse.io",
  "Template": "daily-shift-reports",
  "TemplateData": "{\\"activeShifts\\":[{\\"id\\":\\"565fe59257c5d59e60cc35a2\\",\\"link\\":\\"http://link.com\\"}],\\"date\\":\\"Wednesday 6th November\\",\\"hasNoShifts\\":false,\\"isProduction\\":true,\\"name\\":\\"Steve\\",\\"shortDate\\":\\"November 6\\",\\"stage\\":\\"PRODUCTION\\"}",
}
`;

exports[`helpers:buildShiftEmail it returns email data ordered by lastName and Timestamp 1`] = `
Object {
  "Destination": Object {
    "CcAddresses": Array [],
    "ToAddresses": Array [
      "testing@test.com",
    ],
  },
  "ReplyToAddresses": Array [
    "notifications@lighthouse.io",
  ],
  "Source": "notifications@lighthouse.io",
  "Template": "daily-shift-reports",
  "TemplateData": "{\\"activeShifts\\":[{\\"id\\":\\"565fe59257c5d59e60cc35a6\\",\\"link\\":\\"http://link.com\\",\\"fullName\\":\\"Paul Allen\\",\\"timestamp\\":null,\\"lastName\\":\\"Allen\\"},{\\"id\\":\\"565fe59257c5d59e60cc35a6\\",\\"link\\":\\"http://link.com\\",\\"fullName\\":\\"John Doe\\",\\"timestamp\\":null,\\"lastName\\":\\"Doe\\"},{\\"id\\":\\"565fe59257c5d59e60cc35a6\\",\\"link\\":\\"http://link.com\\",\\"fullName\\":\\"Steeve Jobs\\",\\"timestamp\\":null,\\"lastName\\":\\"Jobs\\"}],\\"date\\":\\"Wednesday 6th November\\",\\"hasNoShifts\\":false,\\"isProduction\\":true,\\"name\\":\\"Steve\\",\\"shortDate\\":\\"November 6\\",\\"stage\\":\\"PRODUCTION\\"}",
}
`;

exports[`helpers:buildShiftEmail it returns email data with configuration set when the environment var is set 1`] = `
Object {
  "ConfigurationSetName": "ses-configuration-set",
  "Destination": Object {
    "CcAddresses": Array [],
    "ToAddresses": Array [
      "testing@test.com",
    ],
  },
  "ReplyToAddresses": Array [
    "notifications@lighthouse.io",
  ],
  "Source": "notifications@lighthouse.io",
  "Template": "daily-shift-reports",
  "TemplateData": "{\\"activeShifts\\":[{\\"id\\":\\"565fe59257c5d59e60cc35a2\\",\\"link\\":\\"http://link.com\\"}],\\"date\\":\\"Wednesday 6th November\\",\\"hasNoShifts\\":false,\\"isProduction\\":true,\\"name\\":\\"Steve\\",\\"shortDate\\":\\"November 6\\",\\"stage\\":\\"PRODUCTION\\"}",
}
`;

exports[`helpers:buildShiftEmail it returns email data with formSubmissions 1`] = `
Object {
  "Destination": Object {
    "CcAddresses": Array [],
    "ToAddresses": Array [
      "testing@test.com",
    ],
  },
  "ReplyToAddresses": Array [
    "notifications@lighthouse.io",
  ],
  "Source": "notifications@lighthouse.io",
  "Template": "daily-shift-reports",
  "TemplateData": "{\\"activeShifts\\":[{\\"id\\":\\"565fe59257c5d59e60cc35a6\\",\\"link\\":\\"http://link.com\\",\\"fullName\\":\\"Paul Allen\\",\\"timestamp\\":null,\\"lastName\\":\\"Allen\\"},{\\"id\\":\\"565fe59257c5d59e60cc35a3\\",\\"link\\":\\"activity-reports/shifts/565e42d3d4c628373ab25231/2019/10/10/2019-10-10-shift-5d9f83bd8bb82c0001cc20c3-summary-1570749390473.pdf\\"}],\\"date\\":\\"Wednesday 6th November\\",\\"hasNoShifts\\":false,\\"isProduction\\":true,\\"name\\":\\"Steve\\",\\"shortDate\\":\\"November 6\\",\\"stage\\":\\"PRODUCTION\\"}",
}
`;

exports[`helpers:buildShiftEmail it returns email data without formSubmissions 1`] = `
Object {
  "Destination": Object {
    "CcAddresses": Array [],
    "ToAddresses": Array [
      "testing@test.com",
    ],
  },
  "ReplyToAddresses": Array [
    "notifications@lighthouse.io",
  ],
  "Source": "notifications@lighthouse.io",
  "Template": "daily-shift-reports",
  "TemplateData": "{\\"activeShifts\\":[{\\"id\\":\\"565fe59257c5d59e60cc35a6\\",\\"link\\":\\"http://link.com\\",\\"fullName\\":\\"Paul Allen\\",\\"timestamp\\":null,\\"lastName\\":\\"Allen\\"},{\\"id\\":\\"565fe59257c5d59e60cc35a3\\",\\"link\\":\\"activity-reports/shifts/565e42d3d4c628373ab25231/2019/10/10/2019-10-10-shift-5d9f83bd8bb82c0001cc20c3-summary-and-forms-1570749390473.pdf\\"}],\\"date\\":\\"Wednesday 6th November\\",\\"hasNoShifts\\":false,\\"isProduction\\":true,\\"name\\":\\"Steve\\",\\"shortDate\\":\\"November 6\\",\\"stage\\":\\"PRODUCTION\\"}",
}
`;

exports[`helpers:buildShiftEmail when no active or inactive shift it returns email data with hasNoShifts set to true 1`] = `
Object {
  "Destination": Object {
    "CcAddresses": Array [],
    "ToAddresses": Array [
      "testing@test.com",
    ],
  },
  "ReplyToAddresses": Array [
    "notifications@lighthouse.io",
  ],
  "Source": "notifications@lighthouse.io",
  "Template": "daily-shift-reports",
  "TemplateData": "{\\"activeShifts\\":[],\\"date\\":\\"Wednesday 6th November\\",\\"hasNoShifts\\":true,\\"isProduction\\":true,\\"name\\":\\"Steve\\",\\"shortDate\\":\\"November 6\\",\\"stage\\":\\"PRODUCTION\\"}",
}
`;

exports[`helpers:buildShiftEmail when not running in production it sets isProduction as false, adds stage to subject line and uses test template 1`] = `
Object {
  "Destination": Object {
    "CcAddresses": Array [],
    "ToAddresses": Array [
      "testing@test.com",
    ],
  },
  "ReplyToAddresses": Array [
    "notifications@lighthouse.io",
  ],
  "Source": "notifications@lighthouse.io",
  "Template": "daily-shift-reports-test",
  "TemplateData": "{\\"activeShifts\\":[{\\"id\\":\\"565fe59257c5d59e60cc35a2\\",\\"link\\":\\"http://link.com\\"}],\\"date\\":\\"Wednesday 6th November\\",\\"hasNoShifts\\":false,\\"isProduction\\":false,\\"name\\":\\"Steve\\",\\"shortDate\\":\\"November 6\\",\\"stage\\":\\"TESTING\\"}",
}
`;
